File: for-local-test/python/338.counting-bits.py
from typing import List

class Solution:
    def countBits(self, n: int) -> List[int]:
        res = [0] * (n + 1)
        for i in range(1, n + 1):
            res[i] = res[i & (i - 1)] + 1
        return res
    
if __name__ == "__main__":
    solution = Solution()
    print(solution.countBits(2))  # [0,1,1]
    print(solution.countBits(5))  # [0,1,1,2,1,2]

File: for-local-test/338.counting-bits.java
class Solution {
    public static int[] countBits(int n) {
        int [] res = new int[n + 1];
        for (int i = 1; i <= n; i++){
            res[i] = res[i & (i - 1)] + 1;
        }
        return res;
    }
    public static void main(String[] args){
        int n = 5;
        int[] result = countBits(n);
        for (int i = 0; i < result.length; i++) {
            System.out.print(result[i] + " ");
        }
    }
}


